File: 2021/day1/part2.py

from AOC import read_input

ds = read_input().splitlines()
ds = list(map(int, ds))
ds = [sum(i) for i in zip(ds[:-2], ds[1:-1], ds[2:])]
print([j>i for i, j in zip(ds[:-1], ds[1:])].count(True))

File: 2021/day7/part2.hs
import AOC
import Data.List

summation :: Integer -> Integer
summation n = (n+1) * n `div` 2

computeFuel :: [Integer] -> Integer -> Integer
computeFuel xs x = sum (map (summation . abs . (x-)) xs)

main = do
    inp <- readInput
    let pos = map read (splitOn "," inp) :: [Integer]
    print $ minimum (map (computeFuel pos) [(minimum pos)..(maximum pos)])
